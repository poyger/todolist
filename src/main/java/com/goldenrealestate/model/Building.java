package com.goldenrealestate.model;

import org.apache.wicket.model.IModel;

import java.io.Serializable;

/**
 * User: Poyan Gerami
 * Email: poyan.gerami@eniro.com
 * Date: 16/06/16
 */
public class Building implements Serializable {
    private String name;

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "Building{" +
                "name='" + name + '\'' +
                '}';
    }

    public void setName(String name) {
        this.name = name;
    }
}
