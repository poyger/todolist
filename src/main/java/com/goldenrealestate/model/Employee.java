package com.goldenrealestate.model;

import java.io.Serializable;

/**
 * User: Poyan Gerami
 * Email: poyan.gerami@eniro.com
 * Date: 16/06/16
 */
public class Employee implements Serializable {
    private String name;
    private Integer age;

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Person{" +
                "name='" + name + '\'' +
                ", age=" + age +
                '}';
    }
}
