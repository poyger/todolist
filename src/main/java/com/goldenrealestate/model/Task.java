package com.goldenrealestate.model;


import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import java.io.Serializable;

@Entity
public class Task implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    String taskDescription;

    @ManyToOne
    Building building;

    @ManyToOne
    Employee employee;


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Building getBuilding() {
        return building;
    }

    public void setBuilding(Building building) {
        this.building = building;
    }

    public Employee getEmployee() {
        return employee;
    }

    public void setEmployee(Employee employee) {
        this.employee = employee;
    }

    public String getTaskDescription() {
        return taskDescription;
    }

    public void setTaskDescription(String task) {
        this.taskDescription = task;
    }

    @Override
    public String toString() {
        return "Task{" +
                "building=" + building +
                ", employee=" + employee +
                ", taskDescription='" + taskDescription + '\'' +
                '}';
    }

}
