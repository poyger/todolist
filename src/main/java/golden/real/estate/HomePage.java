package golden.real.estate;

import de.agilecoders.wicket.core.markup.html.bootstrap.html.HtmlTag;
import de.agilecoders.wicket.core.markup.html.bootstrap.image.GlyphIconType;
import de.agilecoders.wicket.core.markup.html.bootstrap.navbar.Navbar;
import de.agilecoders.wicket.core.markup.html.bootstrap.navbar.NavbarButton;
import org.apache.wicket.markup.html.WebPage;
import org.apache.wicket.model.Model;
import org.apache.wicket.request.mapper.parameter.PageParameters;

import static de.agilecoders.wicket.core.markup.html.bootstrap.navbar.NavbarComponents.transform;

public class HomePage extends WebPage {
	private static final long serialVersionUID = 1L;

	public HomePage(final PageParameters parameters) {
		super(parameters);
        add(new HtmlTag("html"));
        add(newNavbar("navbar"));
    }

    protected Navbar newNavbar(String markupId) {
        Navbar navbar = new Navbar(markupId);

        navbar.setPosition(Navbar.Position.TOP);
        navbar.setInverted(true);

        // show brand name
        navbar.setBrandName(Model.of("Golden Real Estate"));

        navbar.addComponents(transform(Navbar.ComponentPosition.LEFT,
                new NavbarButton<Void>(HomePage.class, Model.of("Overview")).setIconType(GlyphIconType.home)
        ));
        return navbar;
    }
}
