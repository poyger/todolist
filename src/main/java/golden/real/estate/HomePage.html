<!DOCTYPE html>
<html wicket:id="html" xmlns:wicket="http://wicket.apache.org">
<head>
    <meta charset="utf-8">
    <title>Apache Wicket &amp; Bootstrap</title>
</head>

<body style="padding-top: 60px">
<a name="top" style="display: none;"></a>
<a class="sr-only" href="#content">Skip navigation</a>

<header wicket:id="navbar" class="bs-docs-nav"></header>

</body>
</html>
